tasks/toolchains/gcc.rake

```diff
@@ -6,6 +6,7 @@ MRuby::Toolchain.new(:gcc) do |conf, _params|
     cc.option_include_path = '-I%s'
     cc.option_define = '-D%s'
     cc.compile_options = '%{flags} -MMD -o %{outfile} -c %{infile}'
+    cc.cxx_compile_flag = '-x c++ -std=c++03'
```

Comment:
I guess this change makes this flag required now?

Is there some way to do this in a backwards compatible way? i.e. default flag which supports both versions